File: lookyswappy-app/database/models/Score.ts
import { Model, Relation } from '@nozbe/watermelondb'
import { field, relation, date } from '@nozbe/watermelondb/decorators'
import type Round from './Round'
import type GamePlayer from './GamePlayer'

export default class Score extends Model {
  static table = 'scores'
  static associations = {
    rounds: { type: 'belongs_to' as const, key: 'round_id' },
    game_players: { type: 'belongs_to' as const, key: 'player_id' },
  }

  @field('server_id') serverId!: string | null
  @field('round_id') roundId!: string
  @field('player_id') playerId!: string
  @field('raw_score') rawScore!: number
  @field('bonus_applied') bonusApplied!: number
  @field('final_score') finalScore!: number
  @field('total_after') totalAfter!: number
  @date('last_modified') lastModified!: Date
  @field('is_synced') isSynced!: boolean

  @relation('rounds', 'round_id') round!: Relation<Round>
  @relation('game_players', 'player_id') player!: Relation<GamePlayer>
}

File: lookyswappy-api/app/models/score.py
from sqlalchemy import Boolean, ForeignKey, Integer, String
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.models.base import SyncableBase


class Score(SyncableBase):
    """Score for a player in a specific round."""

    __tablename__ = "scores"

    round_id: Mapped[str] = mapped_column(
        String(36), ForeignKey("rounds.id"), nullable=False
    )
    player_id: Mapped[str] = mapped_column(
        String(36), ForeignKey("game_players.id"), nullable=False
    )
    raw_score: Mapped[int] = mapped_column(Integer, nullable=False)
    bonus_applied: Mapped[bool] = mapped_column(Boolean, default=False)
    final_score: Mapped[int] = mapped_column(Integer, nullable=False)

    # Relationships
    round: Mapped["Round"] = relationship("Round", back_populates="scores")  # noqa: F821
    player: Mapped["GamePlayer"] = relationship(  # noqa: F821
        "GamePlayer", back_populates="scores"
    )
